app/poi_api/poi_service.py

```diff
@@ -3,102 +3,51 @@
 from .poi_suggestion import POI_suggestions
 from .poi import POI
 from .poi_errors import POINotFoundError, InvalidPOISuggestionError
+from app import common
 from datetime import datetime
 
 # TODO: Fix error handling
 class POI_Service:
+    def __init__(self):
+        self.poi_ref = firestore_db.collection("POI")
+        self.poi_suggestion_ref = firestore_db.collection("POI_proposal")
+
     def get_all_POI(self):
-        try:
-            poi_ref = firestore_db.collection("POI")
-            all_poi = [doc.to_dict() for doc in poi_ref.stream()]
-            list_all_poi = []
-            for poi in all_poi:
-                poi_object = self._create_poi_object(poi)
-                list_all_poi.append(poi_object)
-            return list_all_poi
-        except Exception as e:
-            raise Exception(f"An error occured: {e}")
+        return [
+            POI.from_dict(self.poi_ref, doc.to_dict()) for doc in self.poi_ref.stream()
+        ]
 
     def get_POI(self, poi_id):
-        try:
-            poi_ref = firestore_db.collection("POI")
-            poi = poi_ref.document(poi_id).get()
-            poi_dict = poi.to_dict()
-            if poi_dict is not None:
-                return self._create_poi_object(poi_dict)
-            else:
-                raise POINotFoundError(
-                    f"The POI with the id: {poi_id} could not be found", 404
-                )
-        except POINotFoundError as e:
-            raise POINotFoundError(
-                f"The POI with the id: {poi_id} could not be found", 404
-            )
-        except Exception as e:
-            raise Exception(f"An error occured: {e}", 404)
+        return POI.get(self.poi_ref, poi_id)

```

Comment:
Yeah in my opinion it's really not a good idea to move all this logic into the data class. It should stay as pure as possible.